shared/src/ledger/native_vp/masp.rs

```diff
@@ -161,6 +322,18 @@ where
                     return Ok(false);
                 }
             }
+
+            if !(self.valid_spend_descriptions_anchor(&shielded_tx)?
+                && self.valid_nullifiers_reveal(keys_changed, &shielded_tx)?)
```

Comment:
In case `source != Address::Internal(Masp)`, should we add a check to verify that no `shielded_spends` are attached to the `Transaction`?

Reply:
Yes, this sounds good to me, thanks! (Maybe you can also add this to the `Boundary Conditions` section in https://specs.namada.net/masp/ledger-integration ?)

Reply:
For simplicity's sake I'm addressing this in #2248